exercises/dolica_akelloegwel/ex02_oo_basics/src/main.cpp

```diff
@@ -25,7 +25,7 @@ int main(int, char **)
 	Triangle myTriangle = Triangle(triangleBase, triangleHeight);
 
 	// Create a vector and place the shapes in it
-	std::vector<Shape> myShapes;
+	std::vector<std::reference_wrapper<Shape> > myShapes;

```

Comment:
Your `reference_wrapper` is safe, as currently written, and an interesting approach, but could result in unusual/undefined behaviour with some minor modification. Try storing the `Shape` pointers as `std::unique_ptr<Shape>` ask me for help if you have any questions.